### PR TITLE
[Communication] - Common - Update CommunicationTokenCredential ctor to use CommunicationTokenRefreshOptions

### PR DESCRIPTION
**Pull request by minnieliu**

### CODE REVIEW

**Comment by JianpingChen** [comment]
Please update changelog for the breaking changes: removing constructors

File: sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationTokenRefreshOptions.java [hunk 1]
```diff
@@ -47,15 +47,12 @@ public CommunicationTokenRefreshOptions(Callable<String> tokenRefresher, boolean
...
-     * @param token the serialized JWT token, cannot be null
+     * @param initialToken the serialized JWT token, cannot be null
      */
     public CommunicationTokenRefreshOptions(Callable<String> tokenRefresher, boolean refreshProactively, String initialToken) {
         if (tokenRefresher == null) {
             throw new IllegalArgumentException("Missing required parameters 'tokenRefresher'.");
         }
-        if (initialToken == null) {
-            throw new IllegalArgumentException("Missing required parameters 'initialToken'.");
-        }
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationTokenRefreshOptions.java:58]: Please keep this. If initialToken is not available, caller should be using the other constructor

**Review by JianpingChen — CHANGES_REQUESTED** [review]

**Review by jbeauregardb — APPROVED** [review]
In addition to JP's comment, we should also update the README. Other than that LGTM :)

**Review by RezaJooyandeh — APPROVED** [review]
Thank you for the quick update

File: sdk/communication/azure-communication-common/README.md [hunk 2]
```diff
@@ -94,31 +94,31 @@ Callable<String> tokenRefresher = () -> {
...
 ```
 
 ### Creating a credential that refreshes proactively
 
-Setting `isProactiveRefresh` to true will call your `Callable<String> tokenRefresher` when the token is close to expiry.
+Setting `refreshProactively` to true will call your `Callable<String> tokenRefresher` when the token is close to expiry.
 
 ```java
-CommunicationTokenCredential userCredential = new CommunicationTokenCredential(tokenRefresher, true);
+CommunicationTokenCredential userCredential = new CommunicationTokenCredential(new CommunicationTokenRefreshOptions(tokenRefresher, true, null));
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-common/README.md:105]: User the other constructor by remove ", null"

File: sdk/communication/azure-communication-common/README.md [hunk 3]
```diff
@@ -94,31 +94,31 @@ Callable<String> tokenRefresher = () -> {
...
+CommunicationTokenCredential userCredential = new CommunicationTokenCredential(new CommunicationTokenRefreshOptions(tokenRefresher, true, "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjM2MDB9.adM-ddBZZlQ1WlN3pdPBOF5G4Wh9iZpxNP_fSvpF4cWs"));
 ```
 
 ### Getting a token asynchronously
 
 Calling `getToken()` will return a `Future<AccessToken>`
 
 ```java
-CommunicationTokenCredential userCredential = new CommunicationTokenCredential(tokenRefresher);
+CommunicationTokenCredential userCredential = new CommunicationTokenCredential(new CommunicationTokenRefreshOptions(tokenRefresher, false, null));
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-common/README.md:121]: Similarly here.

File: sdk/communication/azure-communication-common/README.md [hunk 4]
```diff
@@ -94,31 +94,31 @@ Callable<String> tokenRefresher = () -> {
   return fetchtoken();
 };
 
-CommunicationTokenCredential userCredential = new CommunicationTokenCredential(tokenRefresher);
+CommunicationTokenCredential userCredential = new CommunicationTokenCredential(new CommunicationTokenRefreshOptions(tokenRefresher, false, null));
```
Comment (JianpingChen) [inline sdk/communication/azure-communication-common/README.md:97]: User the other constructor by remove ", null"

**Review by JianpingChen — CHANGES_REQUESTED** [review]

**Review by JianpingChen — APPROVED** [review]

**Pull request merged**